src/modules/adaptors/Frax/FraxlendHealthFactorLogic.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.21;
+
+import { Math } from "src/utils/Math.sol";
+import { IFToken } from "src/interfaces/external/Frax/IFToken.sol";
+
+/**
+ * @title Fraxlend Health Factor Logic contract.
+ * @notice Implements health factor logic used by both
+ *         the CollateralFTokenAdaptor && DebtFTokenAdaptor.
+ * @author crispymangoes, 0xEinCodes
+ * NOTE: helper functions made virtual in case future Fraxlend Pair versions require different implementation logic. The logic here is written in compliance with FraxlendV2; specifically the helper `_toBorrowAmount()`
+ * NOTE: v2 Debt and Collateral adaptors inherit this contract. v1 adaptors inherit this ultimately, and overrides `_toBorrowAmount()`.
+ */
+contract FraxlendHealthFactorLogic {
+    using Math for uint256;
+
+    /**
+     * @notice The ```_isSolvent``` function returns the current health factor of a respective position given an exchange rate
+     * @param _fraxlendPair The specified Fraxlend Pair
+     * @param _exchangeRate The exchange rate, i.e. the amount of collateral to buy 1e18 asset
+     * @return currentHF The health factor of the position atm
+     */
+    function _isSolvent(IFToken _fraxlendPair, uint256 _exchangeRate) internal view virtual returns (uint256) {
+        // calculate the borrowShares
+        uint256 borrowerShares = _fraxlendPair.userBorrowShares(address(this));
+        uint256 _borrowerAmount = _toBorrowAmount(_fraxlendPair, borrowerShares, true, true); // need interest-adjusted and conservative amount (round-up) similar to `_isSolvent()` function in actual Fraxlend contracts.
+        if (_borrowerAmount == 0) return 1.05e18;
+        uint256 _collateralAmount = _fraxlendPair.userCollateralBalance(address(this));
+        if (_collateralAmount == 0) return 0;
+
+        (uint256 LTV_PRECISION, , , , uint256 EXCHANGE_PRECISION, , , ) = _fraxlendPair.getConstants();
```

Comment:
Could make a `_getConstants` helper, just in case they add more constants in the future